common/src/main/java/net/opentsdb/core/Registry.java

```diff
@@ -74,6 +75,9 @@ public void registerPlugin(final Class<?> clazz,
    */
   public <T> T getPlugin(final Class<T> clazz, final String id);
 
+  /** @return An umodifiable map of registered plugins. */
+  public Map<Class<?>, Map<String, TSDBPlugin>> plugins();
```

Comment:
```suggestion
  public Map<Class<? extends TSDBPlugin>, Map<String, TSDBPlugin>> plugins();
```

Reply:
good point will have to tighten it up under the hood a bit later